src/tests/fuzz/fuzz_decodebase64.c

```diff
@@ -0,0 +1,55 @@
+/**
+ * @file
+ * @author Shivani Bhardwaj <shivani@oisf.net>
+ * fuzz target for DecodeBase64
+ */
+
+#include "suricata-common.h"
+#include "suricata.h"
+#include "util-base64.h"
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size);
+
+static int initialized = 0;
+
+static void ConstrainedDestinationSizeTest(const uint8_t *src, size_t len)
+{
+    for (uint8_t mode = BASE64_MODE_RELAX; mode <= BASE64_MODE_RFC4648; mode++) {
+        uint32_t dest_size = 3 * len / 4;
```

Comment:
I think it is `3 * ((len + 3)/ 4)`

Check with len = 1 ;-)

Reply:
Indeed. Missing a very apparent edge case.

Reply:
Wait. We want to constrain the space here. I think I kept this here this way because dest_size is controlled by the callers. So, what if the caller has run out of space in the dest buffer but still has a byte to decode? Would it not still make a valid test case?

Reply:
Does Suricata want to partially decode base64 ?

Reply:
yes, in case we run out of space to decode more data, I think. wdyt?

Reply:
So, `dest_size` should be random from the input ?